src/WaterParam.ts

```diff
@@ -0,0 +1,131 @@
+import * as PIXI from "pixi.js";
+
+/**
+ * Class to model a water quality parameter.
+ *
+ * @param name - Display name of the parameter.
+ * @param key - Given key of the parameter in the NN-model.
+ * @param value - Initial value of the parameter.
+ * @param increment - The increment with which the value changes.
+ * @param minValue - Minimum value the parameter can be (default 0)
+ * @param maxValue - Maximum value the parameter can be (default 100)
+ * @param optimalMinValue - Minimum value that is considered 'optimal' (default 40)
+ * @param optimalMaxValue - Maximum value that is considered 'optimal' (default 60)
+ *
+ * @example Initializing new Parameter
+ * ```ts
+ *new WaterParam("Parameter A", "parameter_a", 10, 1, 0, 100, 40, 60);
+ *new WaterParam("Parameter B", "parameter_b", 10, 1);
+ *```
+ */
+export class WaterParam {
+  public name: string; // displayname of parameter
+  private _keyName: string; // name of parameter in model
+  private _value: number; // initial value of parameter
+  private _increment: number; // value with which to change the param's value per step (*1)
+  private _minValue: number; // minimum value this parameter can be.
+  private _maxValue: number; // maximum value (inclusive) this parameter can be.
+  private _optimalMinValue: number; // minimum value for the optimal values range.
+  private _optimalMaxValue: number; // maximum value for the optimal values range.
+
+  constructor(
+    name: string,
+    keyName: string,
+    value: number,
+    increment: number,
+    minValue: number = 0,
+    maxValue: number = 100,
+    optimalMinValue: number = 40,
+    optimalMaxValue: number = 60
+  ) {
+    this.name = name;
+    this._minValue = minValue;
+    this._maxValue = maxValue;
+    this._optimalMinValue = optimalMinValue;
+    this._optimalMaxValue = optimalMaxValue;
+    this._keyName = keyName;
+    this._increment = increment;
+    this.value = value;
+
+    // sanity check DEBUG ONLY
+    console.log(
+      `WaterParam created: ${this.name} (${this.keyName}), value: ${this.value}, increment:${this.increment}`
```

Comment:
moet dit eruit?

Reply:
heb wel meerdere console.logs erin staan volgens mij. dioe comment is alleen wel heel enthousiast